File: ios/MobileTracker/Configuration.swift
import Foundation

/// SDK configuration settings
public class Configuration {
    /// API key for authentication with the backend
    public let apiKey: String
    
    /// Backend endpoint URL where events are sent
    public let endpoint: String
    
    /// Maximum number of events to store in the queue
    public let maxQueueSize: Int
    
    public init(apiKey: String, endpoint: String, maxQueueSize: Int = 100) {
        self.apiKey = apiKey
        self.endpoint = endpoint
        self.maxQueueSize = maxQueueSize
    }
}

File: android/src/main/java/com/mobiletracker/Configuration.kt
package com.mobiletracker

/**
 * Configuration storage for the Mobile Tracking SDK
 * 
 * @property apiKey The API key for authenticating with the backend
 * @property endpoint The backend server URL where events are sent
 * @property maxQueueSize Maximum number of events to store in the queue (default: 100)
 */
data class Configuration(
    val apiKey: String,
    val endpoint: String,
    val maxQueueSize: Int = 100
)
